src/Kestrel.Core/Internal/Http/HttpProtocol.cs

```diff
@@ -605,19 +596,13 @@ public void OnHeader(Span<byte> name, Span<byte> value)
                     await FireOnCompleted();
                 }
 
-                if (badRequestException != null)
-                {
-                    // Handle BadHttpRequestException thrown during app execution or remaining message body consumption.
-                    // This has to be caught here so StatusCode is set properly before disposing the HttpContext
-                    // (DisposeContext logs StatusCode).
-                    SetBadRequestState(badRequestException);
-                }
-
                 application.DisposeContext(httpContext, _applicationException);
 
-                // StopStreams should be called before the end of the "if (!_requestProcessingStopping)" block
-                // to ensure InitializeStreams has been called.
-                StopStreams();
+                // Even for non-keep-alive requests, try to consume the entire body to avoid RSTs.
```

Comment:
So we're consuming the body for non keep alive requests? Didn't we do this before and moved away from it?

Reply:
Not writing the response prior to consuming the body could lead to hangs with some clients. Not consuming the entire body leads to another problem where some clients (e.g. HttpClient) will treat the request as failed if it receives and RSTs.

Reply:
Anyway, this isn't functionally different from where we called it before. It's still after writing the response, but now it also after disposing the HttpContext so Hosting's request timing logs are now more accurate.